mpdev/internal/apply/deployment_manager.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apply
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+// DeploymentManagerAutogenTemplate generates a deployment manager template
+// given an autogen.yaml file.
+type DeploymentManagerAutogenTemplate struct {
+	ResourceShared
+	AutogenFile string
+	PartnerID   string `json:"partnerId"`
+	SolutionID  string `json:"solutionId"`
+
+	outDir string
+}
+
+// Apply generates a deployment manager template from an autogen file.
+func (dm *DeploymentManagerAutogenTemplate) Apply() error {
+	dir, err := ioutil.TempDir("", "autogen")
```

Comment:
This is an interesting, and potentially popular, pattern where output of an apply is stored somewhere temporarily to be used in other resources. This is useful for resources that do not have a corresponding remote metadata registry.

Can we invent a "local" metadata registry: applying this resource will output to this local registry for other resources that reference it to use.